Origin: https://github.com/UbuntuBudgie/applications-menu/pull/6
Author: David Mohammed <fossfreedom@ubuntu.com>
Last-Update: 2026-10-03
Description: [PATCH] Remove deprecated granite async load method - appears to
 improve icon loading speed

---
 budgie-applications-menu/applications-menu/src/Widgets/AppButton.vala | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/budgie-applications-menu/applications-menu/src/Widgets/AppButton.vala b/budgie-applications-menu/applications-menu/src/Widgets/AppButton.vala
index 9fb61b23..406116bc 100644
--- a/budgie-applications-menu/applications-menu/src/Widgets/AppButton.vala
+++ b/budgie-applications-menu/applications-menu/src/Widgets/AppButton.vala
@@ -101,11 +101,12 @@ public class Slingshot.Widgets.AppButton : Gtk.Button {
         app_label.wrap_mode = Pango.WrapMode.WORD_CHAR;
         app_label.set_ellipsize (Pango.EllipsizeMode.END);
 
-        var image = new Granite.AsyncImage.from_gicon_async (app.icon, ICON_SIZE);
-        image.pixel_size = ICON_SIZE;
-        image.margin_top = 9;
-        image.margin_end = 6;
-        image.margin_start = 6;
+        var image = new Gtk.Image.from_gicon (app.icon, ICON_SIZE) {
+            margin_top = 9,
+            margin_end = 6,
+            margin_start = 6,
+            pixel_size = ICON_SIZE
+        };
 
         badge = new Gtk.Label ("!");
         badge.visible = false;
@@ -162,7 +163,7 @@ public class Slingshot.Widgets.AppButton : Gtk.Button {
         update_badge_count ();
 #endif
 
-        app.notify["icon"].connect (() => image.set_from_gicon_async.begin (app.icon, ICON_SIZE));
+        app.bind_property("icon", image, "gicon");
     }
 
     public void launch_app () {
-- 
2.55.0

